src/cpu_features/x86.rs

```diff
@@ -35,7 +36,9 @@ impl CpuFeatureLevel {
 
 impl Default for CpuFeatureLevel {
   fn default() -> CpuFeatureLevel {
-    let detected: CpuFeatureLevel = if is_x86_feature_detected!("avx2") {
+    let detected: CpuFeatureLevel = if is_x86_feature_detected!("avx512f") {
```

Comment:
With AVX512, Intel decided to divide features among many cpuflags. You are going to have to check more cpu flags if you want something useful.

You can see which flags dav1d uses in the comments here.
https://code.videolan.org/videolan/dav1d/commit/40891aab9bc2f60bdbef16d1a499f36944e1738d#0dc601e13d41edea90a29ed0c5a00c738474626d_854_848

Wikipedia can also give you the info you need. https://en.wikipedia.org/wiki/AVX-512

Reply:
dav1d basically checks for icelake in the detection routine, I updated our check accordingly, I might need your help to add the missing feature bits in stdarch if we want to be thorough.

Reply:
I added the checks, even if even a partial match should be fine for now at least :)